monai/losses/dice.py

```
@@ -163,27 +113,42 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        if self.ignore_index is not None and valid_mask is None:
+            if 0 <= self.ignore_index < target.shape[1]:
+                valid_mask = torch.ones_like(target)
+                valid_mask[:, self.ignore_index] = 0.0
+
         if not self.include_background:
             if n_pred_ch == 1:
                 warnings.warn("single channel prediction, `include_background=False` ignored.", stacklevel=2)
             else:
-                # if skipping background, removing first channel
                 target = target[:, 1:]
                 input = input[:, 1:]
```

Ruff (0.15.20)

[error] Variable `input` is shadowing a Python builtin

(A001)